src/core/utils/fileutils.cpp

```diff
@@ -373,8 +373,12 @@ void FileUtils::addImageMetadata( const QString &imagePath, const GnssPositionIn
     metadata["Exif.GPSInfo.GPSSpeedRef"] = "K";
   }
 
-  metadata["Exif.GPSInfo.GPSDateStamp"] = positionInformation.utcDateTime().date();
-  metadata["Exif.GPSInfo.GPSTimeStamp"] = positionInformation.utcDateTime().time();
+  // slight change in behaviour, but more correct
+  if ( positionInformation.utcDateTime().isValid() )
+  {
+    metadata["Exif.GPSInfo.GPSDateStamp"] = positionInformation.utcDateTime().date();
+    metadata["Exif.GPSInfo.GPSTimeStamp"] = positionInformation.utcDateTime().time();
+  }
```

Comment:
I see what you're doing, but elsewhere you are using system time instead of gnss time if gnss time is not valid.   Why is this different?  I'm not saying you don't have a reason  but that reason is what should be in the comment.

Are you finding that there is a valid position but not valid time in gnss data?